utils/next_date.go

```diff
@@ -0,0 +1,80 @@
+package utils
+
+import (
+	"fmt"
+	"log"
+	"strconv"
+	"strings"
+	"time"
+)
+
+// NextDate вычисляет следующую дату задачи.
+// now – переданное время (из параметра now запроса), dateStr – исходная дата задачи,
+// repeat поддерживает два формата: "d N" – для ежедневного повторения и "y" – для ежегодного.
+func NextDate(now time.Time, dateStr, repeat string) (string, error) {
+	log.Printf("DEBUG (NextDate): now=%s, dateStr=%s, repeat=%s", now.Format("20060102"), dateStr, repeat)
+	if repeat == "" {
+		return "", fmt.Errorf("правило повторения не указано")
+	}
+
+	// Обязательно используем UTC для устранения проблем с часовыми поясами
+	now = now.UTC().Truncate(24 * time.Hour)
+
+	date, err := time.Parse("20060102", dateStr)
+	if err != nil {
+		return "", fmt.Errorf("некорректная дата: %v", err)
```

Comment:
Для оборачивания используем `%w` (wrap)
```suggestion
		return "", fmt.Errorf("некорректная дата: %w", err)
```